azurerm/internal/services/iottimeseriesinsights/resource_iot_time_series_insights_access_policy.go

```diff
@@ -0,0 +1,212 @@
+package iottimeseriesinsights
+
+import (
+	"fmt"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"regexp"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/timeseriesinsights/mgmt/2018-08-15-preview/timeseriesinsights"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/iottimeseriesinsights/parse"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmIoTTimeSeriesInsightsAccessPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmIoTTimeSeriesInsightsAccessPolicyCreateUpdate,
+		Read:   resourceArmIoTTimeSeriesInsightsAccessPolicyRead,
+		Update: resourceArmIoTTimeSeriesInsightsAccessPolicyCreateUpdate,
+		Delete: resourceArmIoTTimeSeriesInsightsAccessPolicyDelete,
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.TimeSeriesInsightsAccessPolicyID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`^[-\w\._\(\)]+$`),
+					"IoT Time Series Insights Access Policy name must contain only word characters, periods, underscores, and parentheses.",
```

Comment:
The regex also says hyphens are allowed, could we please also update the message accordingly? Or please confirm whether the hyphens are allowed?